src/app/test/test.component.ts

```diff
@@ -65,106 +66,94 @@ export class TestComponent implements OnInit, AfterViewInit {
     private tabService: TabService,
     private errorHandler: ErrorHandling,
   ) {
-    this.getGeneratorStatus();
+    this.setStorageIdsFromLocalStorage();
+    this.setGeneratorStatusFromLocalStorage();
   }
 
-  ngAfterViewInit(): void {
+  ngOnInit(): void {
     this.loadData('');
   }
 
-  openCloneModal(): void {
-    if (this.helperService.getSelectedReports(this.reports).length === 1) {
-      this.cloneModal.open(this.helperService.getSelectedReports(this.reports)[0]);
-    } else {
-      this.toastService.showWarning('Make sure exactly one report is selected at a time');
-    }
-  }
-
-  openSettingsModal(): void {
-    this.testSettingsModal.open();
-  }
-
-  showStorageIds(): boolean {
-    return localStorage.getItem('showReportStorageIds') === 'true';
+  setStorageIdsFromLocalStorage(): void {
+    this.showStorageIds = localStorage.getItem('showReportStorageIds') === 'true';
   }
 
-  ngOnInit(): void {
-    this.loadData('');
-    this.getGeneratorStatus();
-  }
-
-  getGeneratorStatus() {
-    const generatorStatus = localStorage.getItem('generatorEnabled');
+  setGeneratorStatusFromLocalStorage() {
+    const generatorStatus: string | null = localStorage.getItem('generatorEnabled');
     if (generatorStatus) {
-      this.generatorStatus = generatorStatus;
+      this.generatorEnabled = generatorStatus === 'true';
     } else {
       this.httpService.getSettings().subscribe({
         next: (response: OptionsSettings) => {
-          this.generatorStatus = response.generatorEnabled ? 'Enabled' : 'Disabled';
-          localStorage.setItem('generatorEnabled', this.generatorStatus);
+          console.log(response);
+          this.generatorEnabled = response.generatorEnabled;
+          localStorage.setItem('generatorEnabled', String(this.generatorEnabled));
         },
         error: () => catchError(this.errorHandler.handleError()),
       });
     }
   }
 
-  addCopiedReports(metadata: any[]): void {
-    const amountAdded: number = metadata.length - this.reports.length;
-    if (amountAdded > 0) {
-      for (let index = this.reports.length; index <= metadata.length - 1; index++) {
-        if (this.matches(metadata[index])) metadata[index].checked = true;
-        this.reports.push(metadata[index]);
-      }
-    }
-  }
-
-  getCopiedReports(): void {
-    this.httpService.getTestReports(this.currentView.metadataNames, this.currentView.storageName).subscribe({
-      next: (response: TestListItem[]) => this.addCopiedReports(response),
-      error: () => catchError(this.errorHandler.handleError()),
-    });
-  }
-
   loadData(path: string): void {
```

Comment:
```suggestion
  loadData(path?: string): void {
```
Wouldn't it be better to have this as an optional paramater? Currently I only see one place where this method is called where it was not an empty string.

Reply:
If we do this we can remove most of the falsy empty strings from the other method calls giving cleaner, more readable code.

Reply:
Oh right, good one, changed it.